client/setec/client_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package setec_test
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/tailscale/setec/client/setec"
+	"github.com/tailscale/setec/types/api"
+)
+
+const testSecrets = `{
+  "apple": {
+     "secret": {"Version": 1, "Value": "c2F1Y2U="}
+  },
+  "pear": {
+     "unrelated": true
+  },
+  "plum": {
+     "secret": {"Version": 2, "Value": "dGFydGxldA=="}
```

Comment:
UX suggestion: maybe this struct could have both a Value and a TextValue field, where the latter is defined to be Value without base64 encoding?

We do still need the B64 version for generality with binary secrets, but having a second field where plaintext secrets can be provided verbatim would make this much more usable for 90% of needs without additional tooling.

Fine as a followup though, and maybe not worth the new confusing precedence semantics and so forth?...

Reply:
Another schema option: a `Format` field, with valid values `plain` and `base64`, and "" implying `base64` for backwards compatibility? A fair bit more typing, but maybe better by being more explicit? :shrug:

Reply:
Good ideas. I think I have a mild preference for the first (Value/TextValue), but either would be fine if you prefer the latter.

Reply:
Updated, and some additional tests. PTAL.